src/test/regress/sql/upgrade_pg_dist_cleanup_before.sql

```diff
@@ -0,0 +1,20 @@
+\set upgrade_test_old_citus_version `echo "$CITUS_OLD_VERSION"`
+SELECT substring(:'upgrade_test_old_citus_version', 'v(\d+)\.\d+\.\d+')::int < 10
+AS upgrade_test_old_citus_version_lt_10_0;
+\gset
+\if :upgrade_test_old_citus_version_lt_10_0
+\else
+\q
```

Comment:
could we leave a comment here on why we are _not_ testing this when the version before an upgrade is citus 10 or greater.

Also, since we patched something in Citus 11.2 for this, are we testing for the correct Citus version here?

Lastly, given we test a version and then do nothing in the if-block, and only have 'code' in the else block, would it make sense to inverse the check we are performing?

Reply:
afaik we have two versions to use in the upgrade tests.
v9.5.0 and v10.1.0
Current patch runs the before test for v9.5 and then runs the after test for the latest version. Since the cleanup record changes were introduced after v9.5, we are able to test it by comparing with v9.5. We could've done it for v10.1.0 as well and probably nothing would change. 

I'm not sure I've entirely understood your comment.
Can you explain which patch have you mentioned?

Reply:
> afaik we have two versions to use in the upgrade tests.
> v9.5.0 and v10.1.0

This is currently the case, but once in a blue moon we might revisit the versions we test upgrades against. Don't assume these will be the only version we ever test against. Our version checks need to make sense, not only discriminate between the two current versions we test against.

> Can you explain which patch have you mentioned?

This PR is changing tests we have added via https://github.com/citusdata/citus/pull/6494 which seems to have been landed in 11.2, therefor it seems strange to me that we don't have to test this if Citus originates from v10.x.y. I would assume an upgrade from 10 to 11.2 would also require this `before` and `after` steps to run. Given we specifically quit the test here if the before version is v10 or above it would be great to add a comment on _why_ setting up is either not possible, or will not trigger the desired conditions we are testing.

Reply:
The check in this file should be changed to a before Citus 11.2 check instead.

P.S. To clarify how we'll revisit the version upgrades we test: We'll never run any upgrade tests for Citus 9.4 or smaller, because those have no overlapping PG version support with the current Citus version. But in the future we'll want to be able upgrade from later versions when we drop PG support for the last version that supported v10.1.0.